sympy/physics/mechanics/joint.py

```diff
@@ -0,0 +1,493 @@
+    parent_axis : Vector, optional
+        Axis fixed in the parent body which aligns with an axis fixed in the
+        child body. The default is x axis in parent's reference frame.
+    child_axis : Vector, optional
+        Axis fixed in the child body which aligns with an axis fixed in the
+        parent body. The default is x axis in child's reference frame.
+
+    Attributes
+    ==========
+
+    name : string
+        The joint's name.
+    parent : Body
+        The joint's parent body.
+    child : Body
+        The joint's child body.
+    coordinates : list
+        List of the joint's generalized coordinates.
+    speeds : list
+        List of the joint's generalized speeds.
+    parent_joint : Point
+        The point fixed in the parent body that represents the joint.
+    child_joint : Point
+        The point fixed in the child body that represents the joint.
+    parent_axis : Vector
+        The axis fixed in the parent frame that represents the joint.
+    child_axis : Vector
+        The axis fixed in the child frame that represents the joint.
+
+    Examples
+    =========
+
+    A single pin joint is created from two bodies and has the following basic
+    attributes:
+
+    >>> from sympy.physics.mechanics import Body, PinJoint
+    >>> parent = Body('P')
+    >>> parent
+    P
+    >>> child = Body('C')
+    >>> child
+    C
+    >>> joint = PinJoint('PC', parent, child)
+    >>> joint
+    PinJoint: PC  parent: P  child: C
+    >>> joint.name
+    'PC'
+    >>> joint.parent
+    P
+    >>> joint.child
+    C
+    >>> joint.parent_joint
+    PC_P_joint
+    >>> joint.child_joint
+    PC_C_joint
+    >>> joint.parent_axis
+    P_frame.x
+    >>> joint.child_axis
+    C_frame.x
+    >>> joint.coordinates
+    [theta_PC(t)]
+    >>> joint.speeds
+    [omega_PC(t)]
+    >>> joint.child.frame.ang_vel_in(joint.parent.frame)
+    omega_PC(t)*P_frame.x
+    >>> joint.child.frame.dcm(joint.parent.frame)
+    Matrix([
+    [1,                 0,                0],
+    [0,  cos(theta_PC(t)), sin(theta_PC(t))],
+    [0, -sin(theta_PC(t)), cos(theta_PC(t))]])
+    >>> joint.child_joint.pos_from(joint.parent_joint)
+    0
+
+    To further demonstrate the use of the pin joint, the kinematics of simple
+    double pendulum that rotates about the Z axis of each connected body can be
+    created as follows.
+
+    >>> from sympy import symbols, trigsimp
+    >>> from sympy.physics.mechanics import Body, PinJoint
+    >>> l1, l2 = symbols('l1 l2')
+
+    First create bodies to represent the fixed ceiling and one to represent
+    each pendulum bob.
+
+    >>> ceiling = Body('C')
+    >>> upper_bob = Body('U')
+    >>> lower_bob = Body('L')
+
+    The first joint will connect the upper bob to the ceiling by a distance of
+    ``l1`` and the joint axis will be about the Z axis for each body.
+
+    >>> ceiling_joint = PinJoint('P1', ceiling, upper_bob,
+    ... child_joint_pos=l1*upper_bob.frame.x,
```

Comment:
```suggestion
    ... child_joint_pos=-l1*upper_bob.frame.x,
```
Setting the vector from the joint to the mass center as a positive quantity is clearer and more standard.

Reply:
Done